gcampus/core/views/water/__init__.py

```diff
@@ -0,0 +1,37 @@
+#  Copyright (C) 2022 desklab gUG (haftungsbeschränkt)
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from django.views.generic import DetailView
+
+from gcampus.core.models.water import Water
+from gcampus.core.views.base import TitleMixin
+from gcampus.core.views.water.list import WaterListView
+
+
+class WaterDetailView(TitleMixin, DetailView):
+    model = Water
+    queryset = Water.objects.all()
```

Comment:
In the template you run `water.measurements.count` for every water. This will trigger a new database lookup for every water item. Instead, you can optimise the query set:

```python
Water.objects.filter(measurements__isnull=False).all()
```

Additionally, you count the measurements related to the water for every entry. This can be prefetched with the following:

`.annotate(measurement_count=Count('measurements'))`

And should then be available as `water.measurement_count`.